### PR TITLE
LP-279 Add i18n text to name page

### PR DESCRIPTION
**Pull request by markpit**
### JIRA link
https://companieshouse.atlassian.net/browse/LP-211

### Change description
Change the text on the Name page to use localisation and add text to the EN and CY locales/translations files

### Work checklist

- [x] Tests added where applicable
- [ ] UI changes look good on mobile
- [ ] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,7 +12,7 @@
     "sonarqube-base-branch": "sonar-scanner",
     "sonarqube-pull-request": "sonar-scanner -D sonar.pullrequest.base=main",
     "start": "node dist/bin/www.js",
-    "dev": "nodemon src/bin/www.ts",
+    "dev": "nodemon --ext ts,njk,json src/bin/www.ts",
```
Comment (markpit) [inline package.json:15]: added file extensions to nodemon args to make it reload when changing njk template or json file (e.g. translations.json)

File: src/presentation/test/integration/localisation.test.ts [hunk 2]
```diff
@@ -9,7 +9,7 @@ import { START_URL } from "../../../presentation/controller/global/Routing";
 
 describe("Localisation tests", () => {
   beforeEach(() => {
-    jest.resetAllMocks();
+    jest.clearAllMocks();
```
Comment (markpit) [inline src/presentation/test/integration/localisation.test.ts:12]: when looking at the localisation tests, we had some previous problems with mocking when using the resetAllMocks (if we changed the way the mocking was done), clearAllMocks was the correct level to use, so just updated this while I remembered.

File: package.json (hunk 1 above)
Comment (markpit) [reply package.json:15]: and included ts files of course.

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: locales/cy/translations.json [hunk 3]
```diff
@@ -1,4 +1,19 @@
...
+        "saveAndContinue" : "WELSH - Save and continue"
+    },
+
+    "namePage" : {
+        "title" : "WELSH - Limited partnership name",
+        "whatIsName" : "WELSH - What is the limited partnership name?",
+        "whatIsNameHint" : "WELSH - Do not enter a name ending, like Limited Partnership or LP. We'll add this for you.",
+        "checkName" : "WELSH - Check the limited partnership name against an existing trademark",
+        "checkNameExistingLP" : "WELSH - You must check the name is not the same as an existing limited partnership on the <a href='#/'>Companies House register (link opens in a new tab)</a>. If it is, we will reject your application.",
+        "checkNameExistingTrademark" : "WELSH - You should also use the <a href='https://www.gov.uk/search-for-trade mark' target='_blank'>trade mark search (link opens in a new tab)</a> to check if your name matches an existing trade mark. If it does, consider choosing a different name. Companies House will not carry out this check for you.",
```
Comment (mattch1) [inline locales/cy/translations.json:12]: This isn't Welsh, is this a placeholder?

File: src/views/name.njk [hunk 4]
```diff
@@ -10,12 +10,12 @@
       
       {% include "includes/csrf_token.njk" %}
 
-      <h1 class="govuk-heading-xl">Limited partnership name</h1>
+      <h1 class="govuk-heading-xl">{{ i18n.namePage.title }}</h1>
```
Comment (mattch1) [inline src/views/name.njk:13]: Where is the i18n value set? Also how?

File: locales/cy/translations.json (hunk 3 above)
Comment (markpit) [reply locales/cy/translations.json:12]: It's a lengthy response to have replied via Slack

File: src/views/name.njk (hunk 4 above)
Comment (markpit) [reply src/views/name.njk:13]: It's a lengthy response to have replied via Slack

**Review by mattch1 — APPROVED** [review]

**Review by dlloyd1-ch — APPROVED** [review]

**Pull request merged**